src/main/java/uk/gov/hmcts/reform/professionalapi/service/impl/ProfessionalUserServiceImpl.java

```diff
@@ -209,4 +212,17 @@ public ModifyUserRolesResponse modifyRolesForUser(UserProfileUpdatedData userPro
                 .body(newUserResponse);
     }
 
+    public void checkUserStatusIsActiveByUserId(String userId) {
+        NewUserResponse newUserResponse = null;
+        Optional<ProfessionalUser> user = professionalUserRepository.findById(UUID.fromString(userId));
+
+        if (user.isPresent()) {
+            newUserResponse = RefDataUtil.findUserProfileStatusByEmail(user.get().getEmailAddress(), userProfileFeignClient);
+        }
+
+        if (newUserResponse == null || !IdamStatus.ACTIVE.name().equalsIgnoreCase(newUserResponse.getIdamStatus())) {
+            throw new AccessDeniedException(ERROR_403_USER_IS_NOT_ACTIVE);
+        }

```

Comment:
should be access denied or bad request?

Reply:
Access denied only

Reply:
but i think in other APis, we use bad request for status not matching.. it should be uniform across APIs..i agree it should be Access Denied but it would mean that the other APis will need to change to return Access Denied error